packages/mom5/package.py

```diff
@@ -312,33 +475,45 @@ def build(self, spec, prefix):
         # ./MOM_compile.csh --type $mom_type --platform spack
         with working_dir(join_path(self.stage.source_path, "exp")):
             build = Executable("./MOM_compile.csh")
-
-            if "+optimisation_report" in self.spec:
-                build.add_default_env("REPORT", "true")
-            if "+restart_repro" in self.spec:
-                build.add_default_env("REPRO", "true")
-
-            # The MOM5 commit d7ba13a3f364ce130b6ad0ba813f01832cada7a2
-            # requires the --no_version switch to avoid git hashes being
-            # embedded in the binary.
-            build(
-                "--type",
-                self.spec.variants["type"].value,
-                "--platform",
-                self._platform,
-                "--no_environ",
-                "--no_version"
-            )
+            if "@access-esm1.5" in self.spec:
+                if "+restart_repro" in self.spec:
+                    build.add_default_env("REPRO", "true")
+                build(
+                    "--type",
+                    self.spec.variants["type"].value,
+                    "--platform",
+                    self._platform,
+                    "--no_environ"
+                )
+            else:
+                if "+optimisation_report" in self.spec:
+                    build.add_default_env("REPORT", "true")
+                if "+restart_repro" in self.spec:
+                    build.add_default_env("REPRO", "true")
+
+                # The MOM5 commit d7ba13a3f364ce130b6ad0ba813f01832cada7a2
+                # requires the --no_version switch to avoid git hashes being
+                # embedded in the binary.
+                build(
+                    "--type",
+                    self.spec.variants["type"].value,
+                    "--platform",
+                    self._platform,
+                    "--no_environ",
+                    "--no_version"
+                )
```

Comment:
```suggestion
            if not "@access-esm1.5" in self.spec:
                build.add_default_arg("--no_version")
            if "+optimisation_report" in self.spec:
                build.add_default_env("REPORT", "true")
            if "+restart_repro" in self.spec:
                build.add_default_env("REPRO", "true")
            # The MOM5 commit d7ba13a3f364ce130b6ad0ba813f01832cada7a2
            # requires the --no_version switch to avoid git hashes being
            # embedded in the binary.
            build(
                "--type",
                self.spec.variants["type"].value,
                "--platform",
                self._platform,
                "--no_environ",
            )
```